src/config/lockfile.py

```diff
@@ -74,3 +106,113 @@ def write(self):
         """Writes the state of this lockfile object to the file."""
         with open(self._path, 'w', encoding='utf-8') as f:
             f.write(json.dumps(self, indent=2))
+
+    def add_entry(self, entry: LockfileEntry, entry_type: Asset):
+        """Adds a specified entry to the lockfile object."""
+        if self.get_entry(entry, entry_type) is None:
+            if type is Asset.MOD:
```

Comment:
This seems awkward and repetitive. Maybe we should have the asset type as a property of the asset itself that determines where it should be downloaded to. The entries in the lockfile don't care what type of asset it is. We only care when it needs to be installed to the computer.

In case there are duplicate string names, we can come up with some other scheme for keying.